### ISSUE TITLE
[BUG]  Navigation failed because page crashed! (Javascript tests)

### ISSUE THREAD

**Issue by gkushang** [microsoft/playwright#4655]
**Context:**
- Playwright Version: playwright@1.6.2
- Operating System: Mac
- Node.js version: v12.19.0
- Browser: Chromium

**Describe the bug**

## Error 1

The Page Crash after Navigation. There are many API calls happens to load the heavy page. I use `waitForNavigation=domcontentloaded` and tried others too but it does not load the page. It needs some time to wait rather than crashing. 

Steps that our test flows go thru,

- Login
- (lot of navigation happens in the background before page loads)
- Click the link -- ERROR (random, 8 out of 10 times) 

- next steps that also fails (always) when switching to another TAB with same error

```
     page.reload: Navigation failed because page crashed!
=========================== logs ===========================
waiting for navigation until "domcontentloaded"
============================================================
Note: use DEBUG=pw:api environment variable and rerun to capture Playwright logs.

      at Connection.sendMessageToServer (node_modules/playwright/lib/client/connection.js:69:15)
      at Proxy.<anonymous> (node_modules/playwright/lib/client/channelOwner.js:54:53)
```

```
  pw:api => page.reload started +1s
  pw:api waiting for navigation until "domcontentloaded" +1ms
  pw:api <= page.reload failed +0ms
  pw:api   "domcontentloaded" event fired +8ms
  pw:api   "load" event fired +0ms
```

## Error 2

If I navigate to another URL in the same tab, I get this error 

```
    [1] <teardown> Error | Error: frame.$: Execution context …
Error: frame.$: Execution context was destroyed, most likely because of a navigation.
```

**Comment by weyert** [comment]
Do you also get this message in Chromium with similar to 'Whoops page crashed' which mentions an error code 101 or similar? I have a similar issue on macOs in Chromium

**Comment by gkushang** [comment]
yeh, the other tab (when I switch to second tab), crashes with the same message.

**Comment by aslushnikov** [comment]
@gkushang can you share a script with us that we can run locally to reproduce this?

**Comment by rinogo** [comment]
@aslushnikov - I just found this Issue via a Google search - I'm not sure if it helps with reproducing/fixing a bug (or if it should be a different Issue altogether), but I get the aforementioned error message when I simply try to load `https://www.californiaorganics.com/`. It looks like it crashes "randomly" at different times - sometimes while loading, other times once loading is complete and my script's actions are underway.

I'm on `1.8.0-next` (`3eef2548e4d55db5970a81582bc0276d814c42da`), macOS 10.15.7, and Node v14.15.1.

```
page.goto: Navigation failed because page crashed!
=========================== logs ===========================
navigating to \"https://www.californiaorganics.com/\", waiting until \"load\"
============================================================
Note: use DEBUG=pw:api environment variable and rerun to capture Playwright logs.
```

**Comment by aslushnikov** [comment]
@rinogo I still cannot reproduce this.. Depending on your environment, this might happen when browser runs out of memory. Do you Would you mind filing a new issue with all the details? 

I'll close this one since it's not much actionable.

**Comment by rinogo** [comment]
@aslushnikov Done! https://github.com/microsoft/playwright/issues/5687

**Comment by MateoSocha** [comment]
Hi. I experienced a problem with the error "page.goto: Navigation failed because page crashed!". I am using the latest version of Playwright (1.23.2) and running on macOS Monterey (12.4). Does anyone know what could be the reason that this error occurs for Safari browsers and for Chrome it works fine?

**Comment by luisariass** [comment]
whats the solutione here?
